File: src/main/java/com/nhan/minisocial/core/repository/NotificationRepository.java
package com.nhan.minisocial.core.repository;

import com.nhan.minisocial.core.entity.NotificationEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface NotificationRepository extends JpaRepository<NotificationEntity, Long> {
    List<NotificationEntity> findAllByToUser(long id);
}

File: services/notification-service-dao/src/main/kotlin/com/nhan/social/notification/repository/NotificationRepository.kt
package com.nhan.social.notification.repository

import com.nhan.social.notification.entity.Notification
import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepositoryBase
import io.quarkus.panache.common.Page
import io.quarkus.panache.common.Sort
import jakarta.enterprise.context.ApplicationScoped
import java.util.UUID

@ApplicationScoped
class NotificationRepository : PanacheRepositoryBase<Notification, UUID> {

    fun findByOwner(ownerId: UUID, page: Int, size: Int): List<Notification> =
        find("ownerId = ?1", Sort.by("createdAt").descending(), ownerId)
            .page(Page.of(page, size))
            .list()

    fun countUnread(ownerId: UUID): Long =
        count("ownerId = ?1 and seen = false", ownerId)

    fun markAllSeen(ownerId: UUID) {
        update("seen = true WHERE ownerId = ?1 and seen = false", ownerId)
    }
}
